tiledb/sm/c_api/tiledb_experimental.h

```diff
@@ -171,6 +174,640 @@ TILEDB_EXPORT int32_t tiledb_array_evolve(
+ *
+ * @param ctx The TileDB context.
+ * @param file The file object.
+ * @param config TileDB Config for setting to create.
+ * @return `TILEDB_OK` for success or `TILEDB_ERR` for error.
+ */
+TILEDB_EXPORT int32_t tiledb_file_store_uri(
+    tiledb_ctx_t* ctx,
+    tiledb_file_t* file,
+    const char*,
+    tiledb_config_t* config);
+
+/**
+ * Store file from VFS File Handle into file array
+ * **Example:**
+ *
+ * @code{.c}
+ * tiledb_file_t* file;
+ * tiledb_file_alloc(ctx, "s3://tiledb_bucket/my_file", &file);
+ * tiledb_file_open(ctx, file, TILEDB_READ);
+ * const char* mime_type;
+ * uint32_t size = 0;
+ * tiledb_file_get_mime_type(ctx, file, *mime_type, &size);
+ * @endcode
+ *
+ * @param ctx The TileDB context.
+ * @param file The file object.
+ * @param config TileDB Config for setting to create.
+ * @return `TILEDB_OK` for success or `TILEDB_ERR` for error.
+ */
+TILEDB_EXPORT int32_t tiledb_file_store_vfs_fh(
+    tiledb_ctx_t* ctx,
+    tiledb_file_t* file,
+    tiledb_vfs_fh_t*,
+    tiledb_config_t* config);
+
+/**
+ * Get the file MIME type
+ *
+ * **Example:**
+ *
+ * @code{.c}
+ * tiledb_file_t* file;
+ * tiledb_file_alloc(ctx, "s3://tiledb_bucket/my_file", &file);
+ * tiledb_file_open(ctx, file, TILEDB_READ);
+ * const char* mime_type;
+ * uint32_t size = 0;
+ * tiledb_file_get_mime_type(ctx, file, *mime_type, &size);
+ * @endcode
+ *
+ * @param ctx The TileDB context.
+ * @param file The file object.
+ * @param mime_type char* to set to mime_type
+ * @param size length of mime string
+ * @return `TILEDB_OK` for success or `TILEDB_ERR` for error.
+ */
+TILEDB_EXPORT int32_t tiledb_file_get_mime_type(
+    tiledb_ctx_t* ctx,
+    tiledb_file_t* file,
+    const char** mime_type,
+    uint32_t size);
+
+/**
+ * Get the file MIME encoding
+ *
+ * **Example:**
+ *
+ * @code{.c}
+ * tiledb_file_t* file;
+ * tiledb_file_alloc(ctx, "s3://tiledb_bucket/my_file", &file);
+ * tiledb_file_open(ctx, file, TILEDB_READ);
+ * const char* mime_encoding;
+ * uint32_t size = 0;
+ * tiledb_file_get_mime_encoding(ctx, file, *mime_type, &size);
+ * @endcode
+ *
+ * @param ctx The TileDB context.
+ * @param file The file object.
+ * @param mime_type char* to set to mime encoding
+ * @param size length of mime string
+ * @return `TILEDB_OK` for success or `TILEDB_ERR` for error.
+ */
+TILEDB_EXPORT int32_t tiledb_file_get_mime_encoding(
+    tiledb_ctx_t* ctx,
+    tiledb_file_t* file,
+    const char** mime_type,
+    uint32_t size);
+
+/**
+ *
+ * **Example:**
+ *
+ * @code{.c}
+ * tiledb_file_t* file;
+ * tiledb_file_alloc(ctx, "s3://tiledb_bucket/my_file", &file);
+ * tiledb_file_open(ctx, file, TILEDB_READ);
+ *
+ * const char* name;
+ * uint32_t size = 0;
+ * tiledb_file_get_original_name(ctx, file, &name, &size);
+ * @endcode
+ *
+ * @param ctx The TileDB context.
+ * @param file The file object.
+ * @return `TILEDB_OK` for success or `TILEDB_ERR` for error.
+ */
+TILEDB_EXPORT int32_t tiledb_file_get_original_name(
```

Comment:
We can add it later, but I believe we need a setter for this too, for the buffer entrypoints.